src/System.Management.Automation/engine/Modules/ScriptAnalysis.cs

```diff
@@ -277,9 +277,18 @@ public override AstVisitAction VisitAssignmentStatement(AssignmentStatementAst a
         //  - Exporting module members
         public override AstVisitAction VisitCommand(CommandAst commandAst)
         {
-            string commandName =
-                commandAst.GetCommandName() ??
-                GetSafeValueVisitor.GetSafeValue(commandAst.CommandElements[0], null, GetSafeValueVisitor.SafeValueContext.ModuleAnalysis) as string;
+            string commandName = commandAst.GetCommandName();
+            if (commandName is null)
+            {
+                try
+                {
+                    commandName = GetSafeValueVisitor.GetSafeValue(commandAst.CommandElements[0], null, GetSafeValueVisitor.SafeValueContext.ModuleAnalysis) as string;
```

Comment:
Could you please add a code comment to document the scenario, maybe with examples of command names we expect here?